### PR TITLE
add swap for windows, add debug for macos, rerender

### PR DESCRIPTION
**Pull request by mattip**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Debug build issues

### CODE REVIEW

**Comment by h-vetinari** [comment]
Could you try removing
https://github.com/conda-forge/ray-packages-feedstock/blob/548b853555ba75c89151eb2bdf9b8da8c705d048/recipe/build-core.sh#L11
?

**Comment by mattip** [comment]
cc @h-vetinari 

This all seems so fragile. The windows builds are newly failing with
```
external/com_google_protobuf/upb/message/internal/message.c(22): error C2099: initializer is not a constant
```

and macOS arm64 is newly failing with
```
In file included from external/zlib/zutil.c:10:
In file included from external/zlib/gzguts.h:20:
In file included from /Users/runner/miniforge3/conda-bld/ray-packages_1730357054105/_build_env/include/c++/v1/stdio.h:108:
/Applications/Xcode_15.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk/usr/include/stdio.h:210:7: error: expected identifier or '('
  210 | FILE    *fdopen(int, const char *) __DARWIN_ALIAS_STARTING(__MAC_10_6, __IPHONE_2_0, __DARWIN_ALIAS(fdopen));
      |          ^
external/zlib/zutil.h:147:33: note: expanded from macro 'fdopen'
  147 | #        define fdopen(fd,mode) NULL /* No fdopen() */
```

It would be nice if we could refactor bazel build to use the conda-provided system libraries instead of building everything from source.

**Comment by mattip** [comment]
Anyhow, to the point of the PR: the macOS x64_86 builds show `CXXFLAGS` has `-stdlib=libc++` [just before the build](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1068274&view=logs&j=1dfee543-4d72-5193-7012-7358e3fc4acd&t=344bb75a-1253-525e-defa-7c6fedd7579f&l=1607):

`CXXFLAGS=-march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -fmessage-length=0 -isystem /Users/runner/miniforge3/conda-bld/ray-packages_1730357074667/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/include -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/ray-packages_1730357074667/work=/usr/local/src/conda/ray-core-2.38.0 -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/ray-packages_1730357074667/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh=/usr/local/src/conda-prefix`

**Comment by mattip** [comment]
> Could you try removing ...

I removed all the bug-fixes I put in over what is needed for a cross-platform build. Let's see which are still needed

File: .azure-pipelines/azure-pipelines-osx.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 jobs:
 - job: osx
   pool:
-    vmImage: macOS-13
+    vmImage: macOS-14
```
Comment (h-vetinari) [inline .azure-pipelines/azure-pipelines-osx.yml:8]: If you want to persist this across rerenders, you can do
```yaml
azure:
  settings_osx:
    pool:
      vmImage: macos-14
```
in `conda-forge.yml`.

Comment (mattip) [reply .azure-pipelines/azure-pipelines-osx.yml:8]: No, it was just an experiment

**Comment by mattip** [comment]
This [blog post](https://www.stevenengelhardt.com/2021/09/22/practical-bazel-depending-on-a-system-provided-c-cpp-library/) provides a recipe for using system libraries in a bazel build.

**Comment by mattip** [comment]
@h-vetinari looking again at the [exploded compilation arguments](https://github.com/conda-forge/ray-packages-feedstock/pull/187#issuecomment-2449546149), I see
```
-isysroot __BAZEL_XCODE_SDKROOT__ \
    -f__BAZEL_XCODE_SDKROOT__/System/Library/Frameworks \
    -f__BAZEL_XCODE_DEVELOPER_DIR__/Platforms/MacOSX.platform/Developer/Library/Frameworks \
```

That seems wrong. I think the `__BAZEL_XCODE_SDKROOT__` is meant to be replaced with an actual path, maybe from the `$SDKROOT` env variable?

**Comment by h-vetinari** [comment]
Yeah that looks suspect. You can try and check if you find something on the bazel-toolchain feedstock related to the SDK that bazel uses. If there's nothing there, then it's likely coming from bazel itself.

**Comment by mattip** [comment]
My bad, I need to use the bazel-toolchain recipe on both arm64 and x86_64 builds. If I recall correctly, on the macOS-12 image, using it on x86_64 broke the build, but on later images it is required.

**Comment by mattip** [comment]
This PR is part of #188, closing.

**Pull request closed without merging**